ipatests/test_integration/test_user_permissions.py

```diff
@@ -65,6 +65,13 @@ def test_delete_preserve_as_alternate_admin(self):
         self.master.run_command(['kinit', self.altadmin],
                                 stdin_text=self.master.config.admin_password)
 
+        # check if username in selinux are not optimized out
+        # related ticket https://pagure.io/SSSD/sssd/issue/3819
+        self.master.run_command(['ssh', '{}@{}'.format(self.altadmin,
```

Comment:
I assume this command is here to have the `semanage` command run as altadmin. I don't think it will work because commands executed by pytests-multihost by default use authentication by ssh key for root user (e.g., look for IPA_ROOT_SSH_KEY ).  If command should run as different user then either the auth part of multihost config needs to be changed  or you can ssh as the user with the right command as param in single call, e.g. https://github.com/freeipa/freeipa/blob/master/ipatests/test_integration/test_legacy_clients.py#L185

On the contrary, I suspect that presence of this command could be the reason for the test timeouts.

In addition, this test is not connected with `test_delete_preserve_as_alternate_admin` in any regard. Adding this test in the middle of the other will make it harder to maintain. It will be better to create a new test method right after `test_delete_preserve_as_alternate_admin` with the intended commands and description.